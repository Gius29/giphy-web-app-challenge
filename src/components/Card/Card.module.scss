.cardContainer {
  img {
    width: 480px;
    margin: 20px;
    max-width: calc(100% - 40px);
  }
}
