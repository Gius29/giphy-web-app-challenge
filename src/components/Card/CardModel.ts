export interface CardModel {
  id: string;
  bitlyUrl: string;
  gifUrl: string;
  gifAlt: string;
}
