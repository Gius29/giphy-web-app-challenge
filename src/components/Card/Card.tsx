import React, { PureComponent } from "react";
import { CardModel } from "./CardModel";

import styles from "./Card.module.scss";

interface StateProps {
  card: CardModel;
}

class Card extends PureComponent<StateProps> {
  render() {
    const { bitlyUrl, gifUrl, gifAlt } = this.props.card;

    return (
      <article className={styles.cardContainer}>
        <a href={bitlyUrl} target="_blank" rel="noreferrer noopener">
          <img src={gifUrl} alt={gifAlt} />
        </a>
      </article>
    );
  }
}

export default Card;
