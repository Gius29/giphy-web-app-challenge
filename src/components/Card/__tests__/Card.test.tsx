import React from "react";
import renderer from "react-test-renderer";
import { CardModel } from "../CardModel";
import Card from "../Card";

const mockCard: CardModel = {
  id: "mockId",
  bitlyUrl: "mockBitlyUrl",
  gifUrl: "mockGifUrl",
  gifAlt: "mockGifAlt"
};

describe("Card", () => {
  it("should renders correctly", () => {
    const result = renderer.create(<Card card={mockCard} />);
    expect(result).toMatchSnapshot();
  });
});
