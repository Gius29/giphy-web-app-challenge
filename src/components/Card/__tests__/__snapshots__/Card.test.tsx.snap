// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should renders correctly 1`] = `
<article
  className="cardContainer"
>
  <a
    href="mockBitlyUrl"
    rel="noreferrer noopener"
    target="_blank"
  >
    <img
      alt="mockGifAlt"
      src="mockGifUrl"
    />
  </a>
</article>
`;
